### ISSUE TITLE
Why do Wpf TextBoxes only mark some Events as Handled

### ISSUE THREAD

**Issue by PxC-MKias** [dotnet/wpf#10138]
### Description

I work on some Interaction Detection and use the fact that Controls mark RoutedEvents as Handled when they deal with it.
Unfortunately, TextBoxes only mark `TextCompositionEventArgs` in the `OnTestInput` method and `ExecutedRoutedEventArgs` in the `OnSpace` as Handled (see [TextEditorTyping.cs#L386](https://github.com/dotnet/wpf/blob/227c5cfddb3b6098d56706c1c22161d1b6dbe88c/src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Documents/TextEditorTyping.cs#L386)). All other Events (Delete, Key Events, ...) that are also  implemented in `TextEditorTyping` are not marked as handled. Is there a Reason for that? It is a problem for my usecase.

### Reproduction Steps

Create Register a ClassHandler for handled Events and debug what Controls handle the events.

```
KeyEventHandler keyEventHandler = /* please ask for the implementation it if its necessary */
EventManager.RegisterClassHandler(typeof(UIElement), UIElement.PreviewKeyDownEvent, keyEventHandler, true);
EventManager.RegisterClassHandler(typeof(UIElement), UIElement.KeyDownEvent, keyEventHandler, true);
EventManager.RegisterClassHandler(typeof(UIElement), UIElement.PreviewKeyUpEvent, keyEventHandler, true);
EventManager.RegisterClassHandler(typeof(UIElement), UIElement.KeyUpEvent, keyEventHandler, true);
```

### Expected behavior

TextBoxes handle all Events that change its text.

### Actual behavior

TextBoxes only mark `TextCompositionEventArgs` in the `OnTestInput` method and `ExecutedRoutedEventArgs` in the `OnSpace` as Handled.
All common KeyEvents do not mark events as Handled, as well as all other event handling methods OnBackspace, OnDelete, OnToggleInsert, ...

 [TextEditorTyping.cs#L386](https://github.com/dotnet/wpf/blob/227c5cfddb3b6098d56706c1c22161d1b6dbe88c/src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Documents/TextEditorTyping.cs#L386)

### Configuration

Not configuration specific as far as I know, but the target framework is net80-windows

**Comment by Symbai** [comment]
> Is there a Reason for that?

Because for example if you handle PreviewKeyDownEvent, the KeyDownEvent is not raised anymore. But also if a parent mousewheel event is handled, subscribers for a child control mousewheel event are not notified as well. Therefore it would be ridiculous if the framework marks all events as handled.

When to mark such an event as handled is described [here](https://learn.microsoft.com/en-us/dotnet/desktop/wpf/events/marking-routed-events-as-handled-and-class-handling?view=netdesktop-9.0#when-to-mark-routed-events-as-handled)

**Comment by miloush** [comment]
That is a good find, Symbai, for reference.

I think the answer is a bit simpler though: The TextBox does not actually handle these events for typing. Typing happens as a result of text composition and that what gets handled.

**Comment by PxC-MKias** [comment]
> Because for example if you handle PreviewKeyDownEvent, the KeyDownEvent is not raised anymore. But also if a parent mousewheel event is handled, subscribers for a child control mousewheel event are not notified as well. Therefore it would be ridiculous if the framework marks all events as handled.
> 
> When to mark such an event as handled is described [here](https://learn.microsoft.com/en-us/dotnet/desktop/wpf/events/marking-routed-events-as-handled-and-class-handling?view=netdesktop-9.0#when-to-mark-routed-events-as-handled)

Thank you for your response. I understand that the PreviewKeyDownEvent is not marked handled. I would expect that KeyDownEvent is handled, but you could argue that it is enough that TextCompositionEventArgs is marked handled. 
As the source you shared states, elements should be marked as handled when setting properties that affect visual representation. In my opinion this includes every event that modifies the text of a TextBox and therefore changes the visual representation.

> Significant responses include setting focus, modifying public state, setting properties that affect visual representation, raising new events, and completely handling an event.

It also seems inconsistent that the OnSpace method marks the event as handled but the OnDelete not.

**Comment by PxC-MKias** [comment]
> That is a good find, Symbai, for reference.
> 
> I think the answer is a bit simpler though: The TextBox does not actually handle these events for typing. Typing happens as a result of text composition and that's what gets handled.

Yes, text composition seems understandable for normal characters. But what is with the commands like Delete/Space, they are also not makred handled consistently
